### PR TITLE
add config, controller, dao, entity, service, utils and tests

### PR DESCRIPTION
**Pull request by ymilenin**

### CODE REVIEW

File: src/main/java/com/onlineshop/controller/BasketController.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            model.addAttribute("allProductInBasket",
+                    basketService.getBasketByUser(user).get().getProducts());
+            model.addAttribute("orderSum", basketService.getSumOfOrderInBasket(user));
+        }
+        return "order";
+    }
+
+    @PostMapping("/order")
+    public String fillOrder(@SessionAttribute("user") User user,
+                            @RequestParam("surname") String surname,
```
Comment (SevostyanovGerman) [inline src/main/java/com/onlineshop/controller/BasketController.java]: optimize

File: src/main/java/com/onlineshop/controller/LoginController.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+   private final UserService userService;
+   private final ProductService productService;
+
+   @Autowired
+   public LoginController(UserService userService, ProductService productService) {
+       this.userService = userService;
+       this.productService = productService;
+   }
+
+   @GetMapping(path = {"/init"})
```
Comment (SevostyanovGerman) [inline src/main/java/com/onlineshop/controller/LoginController.java]: in separate class

File: src/main/java/com/onlineshop/config/RootConfig.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@Configuration
+@PropertySource("classpath:db.properties")
+@EnableTransactionManagement
+@ComponentScans(value = { 
+      @ComponentScan("com.onlineshop.dao"),
+      @ComponentScan("com.onlineshop.service")
+    })
+public class RootConfig {
+
+   private final Environment env;
```
Comment (Sofasmile) [inline src/main/java/com/onlineshop/config/RootConfig.java]: naming

File: src/main/java/com/onlineshop/dao/impl/BasketDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BasketDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void createBasket(User user) {
+        Transaction transaction = null;
```
Comment (Sofasmile) [inline src/main/java/com/onlineshop/dao/impl/BasketDaoImpl.java]: do we need to use transactional here? maybe better use annotation transactional?

**Pull request merged**